modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/internal/client/extension/ClientExtension.java

```diff
@@ -46,13 +46,23 @@ public Map<String, Object> toJSONMap(String pid) {
 
 		Map<String, Object> configMap = new HashMap<>();
 
-		configMap.put("baseURL", "${portalURL}/o/" + projectName);
+		configMap.put(
+			"baseURL",
+			typeSettings.getOrDefault(
+				"baseURL", "${portalURL}/o/" + projectName));
 		configMap.put("description", description);
-		configMap.put("dxp.lxc.liferay.com.virtualInstanceId", "default");
+		configMap.put(
+			"dxp.lxc.liferay.com.virtualInstanceId", virtualInstanceId);
 		configMap.put("name", name);
+		configMap.put(
+			"projectName",
+			typeSettings.getOrDefault("projectName", projectName));
```

Comment:
@rotty3000 are we going to need to adjust the default `baseURL` value based on this value?

Reply:
I'm not sure I understand your comment. Didn't I already do that?

Reply:
Above, you're passing in the default `projectName` value to the default `baseURL` rather than the one from `typeSettings`, so if define a `projectName` in your `yaml` file, the `baseURL` won't be using it. Is that ok?

Reply:
Oh, right right, I actually think I made a mistake here. `projectName` isn't actually settable by users. I'll fix that change.